### ISSUE TITLE
How to print the model?

### ISSUE THREAD

**Issue by lindadamama** [dotnet/TorchSharp#1457]
How to print the model?
I want to see if the model I wrote is consistent with the Python model
The example is as follows:
``
model = SeaFormer(
        cfgs=[model_cfgs['cfg1'], model_cfgs['cfg2'], model_cfgs['cfg3'], model_cfgs['cfg4'], model_cfgs['cfg5']],
        channels=model_cfgs['channels'],
        key_dims=model_cfgs['key_dims'],
        emb_dims=model_cfgs['emb_dims'],
        depths=model_cfgs['depths'],
        num_heads=model_cfgs['num_heads'],
        mlp_ratios=model_cfgs['mlp_ratios'],
        drop_path_rate=model_cfgs['drop_path_rate'])

    input = torch.rand((1, 3, 512, 512))
    print(model)

``
SeaFormer(
  (smb1): StackedMV2Block(
    (stem_block): Sequential(
      (0): Conv2d_BN(
        (c): Conv2d(3, 16, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(16, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (1): ReLU()
    )
    (layer1): InvertedResidual(
      (conv): Sequential(
        (0): Conv2d_BN(
          (c): Conv2d(16, 16, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), groups=16, bias=False)
          (bn): BatchNorm2d(16, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
        (1): ReLU()
        (2): Conv2d_BN(
          (c): Conv2d(16, 16, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(16, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
    )
    (layer2): InvertedResidual(
      (conv): Sequential(
        (0): Conv2d_BN(
          (c): Conv2d(16, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
        (1): ReLU()
        (2): Conv2d_BN(
          (c): Conv2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), groups=64, bias=False)
          (bn): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
        (3): ReLU()
        (4): Conv2d_BN(
          (c): Conv2d(64, 16, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (bn): BatchNorm2d(16, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
    )

**Comment by alinpahontu2912** [comment]
Hey @lindadamama, what do you mean by printhing the model ?

**Comment by lindadamama** [comment]
For instance, when you call `print(model)` in Python code, it will print out the structure of the neural network and the input parameters of the model written in PyTorch. An example is as follows:

<img width="902" alt="Image" src="https://github.com/user-attachments/assets/1043acbe-a587-4fcf-b6e8-5801b45f0f22" />

**Comment by alinpahontu2912** [comment]
Hey @lindadamama, thanks for the fast reply. I believe there is no support for this functionality within torchsharp. For the moment, I suggest you try running the following piece of code and tell me if it suffices: `<your_model>.named_modules().ToList().ForEach(x => Console.WriteLine(x));`
